Application/Frontend/components/FriendsColumn/FriendsTab.tsx

```diff
@@ -17,7 +20,98 @@ import { IconSearch, IconArrowRight } from '@tabler/icons-react';
  * 
  * @returns The JSX element representing the friends tab section, including a search bar and a list of friends.
  */
+
+  // Function to call to go back to the last previous URL
+  function goBack() {
+    window.history.back();
+  }
+
 export function FriendsTab(props: TextInputProps) {
+        const router = useRouter(); // To handle page redirection after the user logs in
+        const [friendList, setFriendList] = useState<string[]>([]);
+
+    // State hooks for form data and validation errors
+    const [formData, setFormData] = useState<{userName: string, password: string}>({
+        userName: "user1",
+        password: "user1Pass",
+      });
+    
+      // To facilitate client-side validation
+      const [usernameError, setUsernameError] = useState(''); // The message for an incorrect username
+      const [passwordError, setPasswordError] = useState(''); // The message for a correct username but incorrect password
+
+  
+      // Redirect you to the Modifying webpage
+      function onModify() {
+        router.push('create-account');
+      }
+    
+      /**
+       * Handles form submission, sending the login request to the server and processing
+       * the response to either proceed to the application or show error messages.
+       *
+       * @param event - The form submission event
+       */
+      const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+    
+        const response = await fetch('/api/login', { // Establishing a promise
```

Comment:
It looks like you are trying to fetch some data from the login to the Friends Tab component to perhaps try and see who is currently logged in to process your query. This is also not the correct way to do this as we would implement an actual authentication provider directly on our login page that would keep track of who is signed in securely, avoiding the need of excessive queries across every single component. I'm assuming these are just placeholders for future processes as I can see that `handleChange` and `handleSubmit` are not used in your component.